roles/lib_openshift/src/lib/service.py

```diff
@@ -0,0 +1,126 @@
+# pylint: skip-file
+# flake8: noqa
+
+
+# pylint: disable=too-many-instance-attributes
+class ServiceConfig(object):
+    ''' Handle service options '''
+    # pylint: disable=too-many-arguments
+    def __init__(self,
+                 sname,
+                 namespace,
+                 ports,
+                 selector=None,
+                 labels=None,
+                 cluster_ip=None,
+                 portal_ip=None,
+                 session_affinity=None,
+                 service_type=None):
+        ''' constructor for handling service options '''
+        self.name = sname
+        self.namespace = namespace
+        self.ports = ports
+        self.selector = selector
+        self.labels = labels
+        self.cluster_ip = cluster_ip
+        self.portal_ip = portal_ip
+        self.session_affinity = session_affinity
+        self.service_type = service_type
+        self.data = {}
+
+        self.create_dict()
+
+    def create_dict(self):
+        ''' return a service as a dict '''
+        self.data['apiVersion'] = 'v1'
+        self.data['kind'] = 'Service'
+        self.data['metadata'] = {}
+        self.data['metadata']['name'] = self.name
+        self.data['metadata']['namespace'] = self.namespace
+        if self.labels:
+            for lab, lab_value  in self.labels.items():
+                self.data['metadata'][lab] = lab_value
+        self.data['spec'] = {}
+
+        if self.ports:
+            self.data['spec']['ports'] = self.ports
+        else:
+            self.data['spec']['ports'] = []
+
+        if self.selector:
+            self.data['spec']['selector'] = self.selector
+
+        self.data['spec']['sessionAffinity'] = self.session_affinity or 'None'
+
+        if self.cluster_ip:
+            self.data['spec']['clusterIP'] = self.cluster_ip
+
+        if self.portal_ip:
+            self.data['spec']['portalIP'] = self.portal_ip
+
+        if self.service_type:
+            self.data['spec']['type'] = self.service_type
+
+# pylint: disable=too-many-instance-attributes,too-many-public-methods
+class Service(Yedit):
```

Comment:
@tbielawa 

This was a design decision that I made when deciding on how I wanted to model these openshift objects.  All of these openshift objects are just basic dictionary data structures.  Yedit is basically a dictionary structure manipulation tool.  It made sense to me to model a service(or any other) oc object on top of a yedit class because it allows you to then set specific paths to items inside of the object's data and build a class around them.

For example (from the PR):
`+    cluster_ip = "spec.clusterIP"`
I can define the path to where attributes live inside of a dictionary and then can code getters/setters on top of it.  Since the data coming out of the oc commands is json or yaml, we can just send the string into the Service(Yedit) class and it can parse and store the data.

I am open to updating the modeling of the classes and I think that we can get more out of the object comparison's when go that route.  This is a longer term discussion and I don't think should block on these.